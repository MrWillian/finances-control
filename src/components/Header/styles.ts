import styled from 'styled-components/native';

export const Container = styled.View`
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  padding: 30px;
  width: 100%;
`;
