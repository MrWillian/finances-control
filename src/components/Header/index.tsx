import React from 'react';
import { TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

import { Container } from './styles';

interface Props {
  navigation ?: any;
}

const Header: React.FC<Props> = ({ navigation }) => {
  
  const openDrawer = () => {
    navigation.openDrawer();
  }

  return (
    <Container>
      <TouchableOpacity onPress={openDrawer}>
        <Icon name="sort" size={22} color="#4F5A7D" />
      </TouchableOpacity>
      <TouchableOpacity>
        <Icon name="person" size={22} color="#4F5A7D" />
      </TouchableOpacity>
    </Container>
  );
}

export default Header;
