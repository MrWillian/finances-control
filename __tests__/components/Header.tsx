import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../../src/components/Header';

it('renders correctly with defaults', () => {
  const button = renderer
    .create(<Header />)
    .toJSON();
  expect(button).toMatchSnapshot();
});
