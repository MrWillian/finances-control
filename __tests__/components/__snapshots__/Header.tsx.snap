// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 30,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 30,
        "width": "100%",
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#4F5A7D",
            "fontSize": 22,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#4F5A7D",
            "fontSize": 22,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
